streams/src/main/java/org/apache/kafka/streams/kstream/SlidingWindowDef.java

```diff
@@ -186,7 +185,7 @@ public void flush() {
             IntegerSerializer intSerializer = new IntegerSerializer();
             ByteArraySerializer byteArraySerializer = new ByteArraySerializer();
 
-            RecordCollector collector = ((ProcessorContextImpl) context).recordCollector();
+            RecordCollector collector = ((RecordCollector.Supplier) context).recordCollector();
```

Comment:
What are the benefits of requiring ProcessorContextImpl implementing the Supplier interface rather than do the casting?

I originally think there will be two categories of users of Kafka Streaming: 1) normal users will only be interfacing the public APIs, 2) some small groups of developers that add windowing def / etc themselves will need to be exposed with the internal implementations in order to code, so it is OK to ask them to know about ProcessorContextImpl. But maybe there are something that I missed here.